internal/controllers/device_data_controller.go

```diff
@@ -407,6 +408,71 @@ func (udc *UserDevicesController) GetUserDeviceErrorCodeQueries(c *fiber.Ctx) er
 	return c.JSON(GetUserDeviceErrorCodeQueriesResponse{Queries: queries})
 }
 
+// ClearUserDeviceErrorCodeQuery godoc
+// @Description Clear user device error codes
+// @Tags        user-devices
+// @Success     200 {object} controllers.QueryDeviceErrorCodesResponse
+// @Security    BearerAuth
+// @Router      /user/devices/{userDeviceID}/error-codes/clear [post]
+func (udc *UserDevicesController) ClearUserDeviceErrorCodeQuery(c *fiber.Ctx) error {
+	userID := helpers.GetUserID(c)
+	udi := c.Params("userDeviceID")
+
+	logger := helpers.GetLogger(c, udc.log)
+
+	userDeviceExists, err := models.UserDevices(
```

Comment:
The middleware will check for ownership, potentially based on anddress, and existence. I would drop this whole thing.